portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,14 +32,21 @@
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int commentsToDisplay = Integer.parseInt(request.getParameter("num-comments"));
     Query query = new Query(commentDatastoreKey);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    Iterator<Entity> iter = results.asIterable().iterator();
+    int idx = 0;
+    while(iter.hasNext() && (idx < commentsToDisplay)) {
```

Comment:
I think doing (idx++ < commentsToDisplay) would also work.

Your choice on if that's more readable or not.

Reply:
I only want to increment idx if a non-empty string is found. Because I'm using an arraylist, some of the entries are empty because I don't trim the arraylist before putting it in the datastore. I should probably ensure that empty entries are not added to the datstore in the first place so I can avoid all of the worrying about finding empty values in my workflows

Reply:
Ah, yeah that would make more sense. But that can happen in a different pr.